File: java/src/main/java/mchehab/com/java/AccelerometerDataFragment.java
package mchehab.com.java;


import android.annotation.SuppressLint;
import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;


public class AccelerometerDataFragment extends Fragment implements SensorEventListener{

    private List<String> listSensorData = new ArrayList<>();
    private SensorManager sensorManager;
    private ArrayAdapter<String> arrayAdapter;


    public AccelerometerDataFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_accelerometer_data, container, false);
        ListView listView = view.findViewById(R.id.listView);
        arrayAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1,
                listSensorData);
        listView.setAdapter(arrayAdapter);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        sensorManager = (SensorManager)getActivity().getApplicationContext().getSystemService
                (Context.SENSOR_SERVICE);
        sensorManager.registerListener(this, sensorManager.getDefaultSensor(Sensor
                .TYPE_ACCELEROMETER), SensorManager.SENSOR_DELAY_NORMAL);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        sensorManager.unregisterListener(this);
    }

    @Override
    public void onSensorChanged(SensorEvent event) {
        if(event.sensor.getType() == Sensor.TYPE_ACCELEROMETER){
            String data = "x: " + event.values[0] + ", y: " + event.values[1] + ", z: " + event
                    .values[2];
            listSensorData.add(data);
            arrayAdapter.notifyDataSetChanged();
        }
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {

    }
}

File: app/src/main/java/mchehab/com/readsensordata/AccelerometerDataFragment.kt
package mchehab.com.readsensordata


import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView


class AccelerometerDataFragment : Fragment(), SensorEventListener {

    lateinit var sensorManager: SensorManager
    lateinit var arrayAdapter: ArrayAdapter<String>
    val sensorData = mutableListOf<String>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_accelerometer_data, container, false)
        arrayAdapter = ArrayAdapter(activity!!, android.R.layout.simple_list_item_1, sensorData)
        val listView = view.findViewById<ListView>(R.id.listView)
        listView.adapter = arrayAdapter
        return view
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        sensorManager = context!!.getSystemService(Context.SENSOR_SERVICE) as SensorManager
        sensorManager.registerListener(this@AccelerometerDataFragment, sensorManager.getDefaultSensor(Sensor
                .TYPE_ACCELEROMETER), SensorManager.SENSOR_DELAY_NORMAL)

    }

    override fun onDetach() {
        super.onDetach()
        sensorManager.unregisterListener(this)
    }

    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {

    }

    override fun onSensorChanged(event: SensorEvent?) {
        if(event?.sensor?.type == Sensor.TYPE_ACCELEROMETER){
            val data = "x: ${event.values[0]}, y: ${event.values[1]}, z: ${event.values[2]}"
            sensorData.add(data)
            arrayAdapter.notifyDataSetChanged()
        }
    }

}
